### ISSUE TITLE
ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider error

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#5971]
VSMac 7.8.0.485
Git revision: 395086ec96857817571c0615b8d7b94abb318d73
Build date: 2018-09-12 17:24:06+00
Build branch: master
Xamarin extensions: 876804254701660cf6ab3573aa56283d9f26b891

With the code shown below in a C# class library if you click at the start or in the 'return' keyword the ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider error is shown in the toolbar

```
using System;
using System.Net;

namespace TestTest
{
	class TestCredentials : ICredentials
	{
		NetworkCredential credential;

		public TestCredentials(string username, string password)
		{
			credential = new                return new NetworkCredential("1", "1");

		}
	}
}
```

Exception:

```
Roslyn error: Extension_Exception ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider : Sequence contains no elements
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00011] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:19 
  at RefactoringEssentials.CSharp.CodeFixes.ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider+<RegisterCodeFixesAsync>d__5.MoveNext () [0x00289] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeFixes/Custom/ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider.cs:60 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.CodeFixes.CodeFixService+<GetCodeFixesAsync>d__16.MoveNext () [0x00094] in /_/src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs:326 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions+<PerformFunctionAsync>d__3`1[T].MoveNext () [0x00043] in /_/src/Workspaces/Core/Portable/ExtensionManager/IExtensionManagerExtensions.cs:95 
ERROR [2018-09-14 14:54:56Z]: Roslyn error: Extension_Exception ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider : Sequence contains no elements
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00011] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:19 
  at RefactoringEssentials.CSharp.CodeFixes.ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider+<RegisterCodeFixesAsync>d__5.MoveNext () [0x00289] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeFixes/Custom/ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider.cs:60 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.CodeFixes.CodeFixService+<GetCodeFixesAsync>d__16.MoveNext () [0x00094] in /_/src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs:326 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions+<PerformFunctionAsync>d__3`1[T].MoveNext () [0x00043] in /_/src/Workspaces/Core/Portable/ExtensionManager/IExtensionManagerExtensions.cs:95 
ERROR [2018-09-14 14:54:59Z]: Roslyn error: Extension_Exception ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider : Sequence contains no elements
  at System.Linq.Enumerable.First[TSource] (System.Collections.Generic.IEnumerable`1[T] source) [0x00011] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-04/external/bockbuild/builds/mono-x64/external/corefx/src/System.Linq/src/System/Linq/First.cs:19 
  at RefactoringEssentials.CSharp.CodeFixes.ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider+<RegisterCodeFixesAsync>d__5.MoveNext () [0x00289] in /Users/vsts/agent/2.140.0/work/1/s/monodevelop/main/external/RefactoringEssentials/RefactoringEssentials/CSharp/CodeFixes/Custom/ReturnMustNotBeFollowedByAnyExpressionCodeFixProvider.cs:60 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.CodeFixes.CodeFixService+<GetCodeFixesAsync>d__16.MoveNext () [0x00094] in /_/src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs:326 
--- End of stack trace from previous location where exception was thrown ---
  at Microsoft.CodeAnalysis.Extensions.IExtensionManagerExtensions+<PerformFunctionAsync>d__3`1[T].MoveNext () [0x00043] in /_/src/Workspaces/Core/Portable/ExtensionManager/IExtensionManagerExtensions.cs:95 
```

> VS bug [#685053](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/685053)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.505 (master)

Pull Request #5977 merged by: Lluis Sanchez
Author: mono
Commit: 55721607097c0317b61597802a58ab76a9b696ec (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1406 (release-7.7)

Pull Request #5985 merged by: Lluis Sanchez
Author: mono
Commit: 0bb6cb6c501529b68357786017b5c15fa89829f5 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #5977 merged by: Lluis Sanchez
Author: mono
Commit: 55721607097c0317b61597802a58ab76a9b696ec (xamarin/monodevelop)
